include/constants/global.h

```diff
@@ -14,6 +14,8 @@
 
 #define PARTY_SIZE 6
 
+#define CATCHING_SHOW_MONS 6
```

Comment:
should this really be in global.h?

Reply:
> should this really be in global.h?

Figured that since this same constant will eventually be used for the GBA Migration app, it made sense to have it readily available